lib/gui/app/modules/image-writer.js

```diff
@@ -296,3 +304,29 @@ exports.flash = (image, drives) => {
     windowProgress.clear()
   })
 }
+
+/**
+ * @summary Cancel write operation
+ * @function
+ * @public
+ *
+ * @example
+ * imageWriter.cancel()
+ */
+exports.cancel = () => {
+  analytics.logEvent('Cancel', {
+    image: selectionState.getImagePath(),
+    drives: selectionState.getSelectedDevices(),
+    uuid: flashState.getFlashUuid(),
+    unmountOnSuccess: settings.get('unmountOnSuccess'),
+    validateWriteOnSuccess: settings.get('validateWriteOnSuccess')
+  })
+
+  if (ipc.server) {
+    const [ socket ] = ipc.server.sockets
+    ipc.server.emit(socket, 'cancel')
+  } else {
+    const error = new Error('Can\'t abort, not connected to IPC server')
+    analytics.logException(error)
+  }
```

Comment:
I think this will be better off in a try / catch block, to also catch situations where `socket` would be null, etc... i.e.